src/Shared/TaskParameter.cs

```diff
@@ -225,6 +259,24 @@ public void Translate(ITranslator translator)
                     translator.Translate(ref stringArrayParam);
                     _wrappedParameter = stringArrayParam;
                     break;
+                case TaskParameterType.Bool:
+                    bool boolParam = _wrappedParameter switch
+                    {
+                        bool hadValue => hadValue,
+                        _ => default,
+                    };
+                    translator.Translate(ref boolParam);
+                    _wrappedParameter = boolParam;
+                    break;
+                case TaskParameterType.Int:
+                    int intParam = _wrappedParameter switch
+                    {
+                        int hadValue => hadValue,
+                        _ => default,
```

Comment:
I think this should be default (and not null) because although you can translate a nullable int, we default to 0 instead if you don't specify it in the task invocation. We don't have to worry about differentiating between required and not required parameters because we have a separate check that every required parameter is present. Correct?

Reply:
`_wrappedParameter` here is an `object`, so when it has an `int` in it, it's actually a boxed int, which can be `null`. But the underlying type is not `int?`, it's `int`, so there's no underlying `null`.

This code is here because of the funky bidirectional nature of `Translate`: the same method here is called for both serialize and deserialize. For serialize, there will always be a value in `_wrappedParameter`, but for deserialize the pointer will start off as `null`, and then be filled in for the first time by this method. The machinery here gets a valid local that can be passed by reference to `Translate(int)`.